### PR TITLE
{AKS} Refactor a sixth part of the create sub-command in acs module

### PR DESCRIPTION
**Pull request by FumingZhang**
**Description**

* Set up service principal profile.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 1]
```diff
@@ -1253,6 +1260,136 @@ def get_enable_ahub(self, enable_validation: bool = False, **kwargs):
...
+        elif secret_intermediate is not None:
+            client_secret = secret_intermediate
+        else:
+            client_secret = secret_raw_value
+
+        # enable_managed_identity
+        # Note: This parameter will not be decorated into the `mc` object.
+        # read the original value passed by the command
+        managed_identity_raw_value = self.raw_param.get(
+            "enable_managed_identity"
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:1315]: make this a local variable?

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 2]
```diff
@@ -1253,6 +1260,136 @@ def get_enable_ahub(self, enable_validation: bool = False, **kwargs):
...
+        if managed_identity_intermediate is not None:
+            enable_managed_identity = managed_identity_intermediate
+        else:
+            enable_managed_identity = managed_identity_raw_value
+
+        # dynamic completion for enable_managed_identity
+        if service_principal and client_secret:
+            enable_managed_identity = False
+            # add to intermediate
+            self.set_intermediate(
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:1332]: looks like this intermediate values are unnecessary.
we can put a lock on the mc or so to remove this.

**Review by andyliuliming — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py (hunk 1 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/decorator.py]: Sure, changed in commit [#8940e46](https://github.com/Azure/azure-cli/pull/19353/commits/8940e464f2dce548ded787e2799bfc881e2b3abc).

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py (hunk 2 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/decorator.py:1345]: Good point, changed in commit [#1cac795](https://github.com/Azure/azure-cli/pull/19353/commits/1cac7959eca08e5208a12b09b9aa44e9f1e01c50).

**Review by andyliuliming — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**